BDArmory/FX/ExplosionFX.cs

```diff
@@ -628,28 +647,45 @@ private void ExecuteBuildingBlastEvent(BuildingBlastHitEvent eventToExecute)
             DestructibleBuilding building = eventToExecute.Building;
-            building.damageDecay = 600f;
+            //building.damageDecay = 600f;
 
-            if (building)
+            if(building && building.IsIntact && !BDArmorySettings.PAINTBALL_MODE)
             {
                 var distanceFactor = Mathf.Clamp01((Range - eventToExecute.Distance) / Range);
-                float damageToBuilding = (BDArmorySettings.DMG_MULTIPLIER / 100) * BDArmorySettings.EXP_DMG_MOD_BALLISTIC_NEW * Power * distanceFactor;
-
-                damageToBuilding *= 2f;
+                float blastMod = 1;
+                switch (ExplosionSource)
+                {
+                    case ExplosionSourceType.Bullet:
+                        blastMod = BDArmorySettings.EXP_DMG_MOD_BALLISTIC_NEW;
+                        break;
+                    case ExplosionSourceType.Rocket:
+                        blastMod = BDArmorySettings.EXP_DMG_MOD_ROCKET;
+                        break;
+                    case ExplosionSourceType.Missile:
+                        blastMod = BDArmorySettings.EXP_DMG_MOD_MISSILE;
+                        break;
+                    case ExplosionSourceType.BattleDamage:
+                        blastMod = BDArmorySettings.EXP_DMG_MOD_BATTLE_DAMAGE;
+                        break;
+                }
+                float damageToBuilding = (BDArmorySettings.DMG_MULTIPLIER / 100) * blastMod * (Power * distanceFactor);
+                damageToBuilding /= 2;
                 damageToBuilding *= BDArmorySettings.BUILDING_DMG_MULTIPLIER;
-
-                building.AddDamage(damageToBuilding);
-
-                if (building.Damage > building.impactMomentumThreshold)
+                //building.AddDamage(damageToBuilding); 
+                building.FacilityDamageFraction += damageToBuilding;
+                //based on testing, I think facilityDamageFraction starts at 100, and demolisheds the building if it hits 0 - which means it will work great as a HP value in the other direction
+                if (building.FacilityDamageFraction > building.impactMomentumThreshold * 2)
                 {
+                    if (BDArmorySettings.DEBUG_DAMAGE) Debug.Log("[BDArmory.ExplosionFX]: Building " + building.name + " demolished due to Explosive damage! Dmg to building: " + building.Damage);
                     building.Demolish();
                 }
                 if (BDArmorySettings.DEBUG_DAMAGE)
                 {
-                    Debug.Log("[BDArmory.ExplosionFX]: Explosion hit destructible building! Hitpoints Applied: " + Mathf.Round(damageToBuilding) +
-                             ", Building Damage : " + Mathf.Round(building.Damage) +
-                             " Building Threshold : " + building.impactMomentumThreshold +
+                    Debug.Log("[BDArmory.ExplosionFX]: Explosion hit destructible building " + building.name + "! Hitpoints Applied: " + Mathf.Round(damageToBuilding) +
```

Comment:
Is there any reason the `damageToBuilding` is rounded here? The value is often pretty small, so `damageToBuilding.ToString("F3")` would probably be better.
Also in line 872 of ProjectileUtils.cs below.

Reply:
Legacy code. It's been converted to formatted strings